unittests/Sema/KeypathFunctionConversionTests.cpp

```diff
@@ -0,0 +1,127 @@
+//===--- UnresolvedMemberLookupTests.cpp --------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "SemaFixture.h"
+#include "swift/AST/GenericParamList.h"
+#include "swift/Sema/ConstraintSystem.h"
+
+using namespace swift;
+using namespace swift::unittest;
+using namespace swift::constraints;
+
+/// Even in the face of a more permissive conversion that might be chosen based
+/// on other ranking rules (e.g., the overload required is non-generic), ensure
+/// that we will select the solution which requires the more narrow conversion
+/// (e.g., the overload *is* generic).
+TEST_F(SemaTest, TestKeypathFunctionConversionPrefersNarrowConversion) {
```

Comment:
@xedin most of this PR remains the same from when you reviewed two years ago but would love your eyes on this unit test I added—very curious to know if I've overcomplicated anything in the programmatic setup of the decls/system.

Reply:
Thanks, no rush! Still gotta send this thru evolution review :)